@kunalaswani Sorry we had a meeting. I found the problem:
```
npm test
	res:
	> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-12233/HPCC-Platform/esp/src
	> run-s lint
	> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-12233/HPCC-Platform/esp/src
	> jshint --config ./.jshintrc ./eclwatch
	eclwatch/XrefDetailsWidget.js: line 121, col 18, Unnecessary semicolon.
	1 error
	npm ERR! code ELIFECYCLE
	npm ERR! errno 2
	npm ERR! eclwatch@1.0.0 lint: `jshint --config ./.jshintrc ./eclwatch`
	npm ERR! Exit status 2
	npm ERR! 
	npm ERR! Failed at the eclwatch@1.0.0 lint script.
	npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
	npm WARN Local package.json exists, but node_modules missing, did you mean to install?
	npm ERR! A complete log of this run can be found in:
	npm ERR!     /mnt/disk1/home/vamosax/.npm/_logs/2019-02-27T16_48_09_789Z-debug.log
	ERROR: "lint" exited with 2.
	npm ERR! Test failed.  See above for more details.
npm test end
```
Usually this kind of error is reported by Smoketest. I will check and fix it soon.

--- a/esp/src/eclwatch/XrefDetailsWidget.js
+++ b/esp/src/eclwatch/XrefDetailsWidget.js
@@ -105,7 +105,20 @@ define([
                     dom.byId(this.id + "LastRun").textContent = params.Modified;
                     dom.byId(this.id + "LastMessage").textContent = params.Status;
                 }
-                this.refreshActionState();
+
+                if (params.Status.indexOf('Generated') !== -1){
+                    this.setDisabled(this.widget._FoundFiles.id, false);
+                    this.setDisabled(this.widget._OrphanFiles.id, false);
+                    this.setDisabled(this.widget._LostFiles.id, false);
+                    this.setDisabled(this.widget._Directories.id, false);
+                    this.setDisabled(this.widget._Errors.id, false);
+                } else {
+                    this.setDisabled(this.widget._FoundFiles.id, true);
+                    this.setDisabled(this.widget._OrphanFiles.id, true);
+                    this.setDisabled(this.widget._LostFiles.id, true);
+                    this.setDisabled(this.widget._Directories.id, true);
+                    this.setDisabled(this.widget._Errors.id, true);
+                }
             },
 
             _onGenerate: function (arg) {
@@ -148,16 +161,5 @@ define([
                 }
             },
 
-            refreshActionState: function () {
-                var disabled = false;
-                if (this.params.Name === "SuperFiles") {
-                    disabled = true;
-                }
-
-                this.foundFilesWidget = registry.byId(this.id + "_FoundFiles").set("disabled", disabled);
-                this.orphanFilesWidget = registry.byId(this.id + "_OrphanFiles").set("disabled", disabled);
-                this.lostFilesWidget = registry.byId(this.id + "_LostFiles").set("disabled", disabled);
-                this.directoriesWidget = registry.byId(this.id + "_Directories").set("disabled", disabled);
-            }
         });
     });

